third_party/terraform/utils/header_transport.go

```diff
@@ -0,0 +1,36 @@
+package google
+
+import (
+	"net/http"
+	"os"
+)
+
+// adapted from https://stackoverflow.com/questions/51325704/adding-a-default-http-header-in-go
+
+type headerTrasportLayer struct {
+	http.Header
+	baseTransit http.RoundTripper
+}
+
+func headerTrasportLayer(baseTransit http.RoundTripper) headerTrasportLayer {
+	if baseTransit == nil {
+		baseTransit = http.DefaultTransport
+	}
+
+	headers := make(http.Header)
+	if requestReason := os.Getenv("CLOUDSDK_CORE_REQUEST_REASON"); v != "" {
+		headers.Set("X-Goog-Request-Reason", requestReason)
```

Comment:
Can we do this in the same spot where we add other header info? I think it would be better to be consistent with other header/useragent information rather than add a new transport wrapper

Reply:
Thought we agreed not to add this header application there because it would not encapsulate those scenarios where its not using the default client or that specific function `sendRequestWithTimeout`. Using a transport layer would properly encapsulate all scenarios

Reply:
resolving after discussion offline. asking for rereview